### PR TITLE
[wasm-mt] Fix pack/build issues in threaded builds

### PR DESCRIPTION
**Pull request by lambdageek**
Addresses the following from https://github.com/dotnet/runtime/issues/74654:

- `dotnet.worker.js` missing from the runtime pack
- [WorkloadManifest.targets.in](https://github.com/dotnet/runtime/blob/536f34d9ab88e153aa11329e789afe6975fbe9a9/src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in#L92) uses `WasmEnableThreading` instead of `WasmEnableThreads`
- The CoreLibs in the multithread and perftrace build variants don't define the correct feature flags.
- `dotnet new wasmbrowser` creates `browser.csproj`, not `<dirname>.csproj`

### CODE REVIEW

File: eng/liveBuilds.targets [hunk 1]
```diff
@@ -188,6 +188,11 @@
         $(LibrariesNativeArtifactsPath)dotnet.timezones.blat;
         $(LibrariesNativeArtifactsPath)*.dat;"
         IsNative="true" />
+      <!-- for threaded wasm -->
+      <LibrariesRuntimeFiles Condition="'$(TargetOS)' == 'Browser' and Exists('$(LibrariesNativeArtifactsPath)dotnet.worker.js')"
+			     Include="
+	$(LibrariesNativeArtifactsPath)dotnet.worker.js"
+	IsNative="true" />
```
Comment (radical) [inline eng/liveBuilds.targets:195]: tabs -> spaces

File: src/mono/wasm/templates/templates/browser/.template.config/template.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
   "name": "WebAssembly Browser App",
   "shortName": "wasmbrowser",
   "sourceName": "browser.0",
+  "preferNameDirectory": true,
```
Comment (radical) [inline src/mono/wasm/templates/templates/browser/.template.config/template.json:9]:
```suggestion
  "preferNameDirectory": true,
```

**Review by radical — APPROVED** [review]

**Comment by lambdageek** [comment]
Not sure why the backport didn't work. it backported cleanly locally. I'll do a manual one in an hour...

**Pull request merged**